node/cmd/guardiand/node.go

```diff
@@ -163,74 +166,77 @@ func init() {
 
-	terraWS = NodeCmd.Flags().String("terraWS", "", "Path to terrad root for websocket connection")
-	terraLCD = NodeCmd.Flags().String("terraLCD", "", "Path to LCD service root for http calls")
-	terraContract = NodeCmd.Flags().String("terraContract", "", "Wormhole contract address on Terra blockchain")
+	// terraWS = NodeCmd.Flags().String("terraWS", "", "Path to terrad root for websocket connection")
+	// terraLCD = NodeCmd.Flags().String("terraLCD", "", "Path to LCD service root for http calls")
+	// terraContract = NodeCmd.Flags().String("terraContract", "", "Wormhole contract address on Terra blockchain")
 
-	algorandIndexerRPC = NodeCmd.Flags().String("algorandIndexerRPC", "", "Algorand Indexer RPC URL")
-	algorandIndexerToken = NodeCmd.Flags().String("algorandIndexerToken", "", "Algorand Indexer access token")
-	algorandAlgodRPC = NodeCmd.Flags().String("algorandAlgodRPC", "", "Algorand Algod RPC URL")
-	algorandAlgodToken = NodeCmd.Flags().String("algorandAlgodToken", "", "Algorand Algod access token")
-	algorandAppID = NodeCmd.Flags().Uint64("algorandAppID", 0, "Algorand app id")
+	// algorandIndexerRPC = NodeCmd.Flags().String("algorandIndexerRPC", "", "Algorand Indexer RPC URL")
+	// algorandIndexerToken = NodeCmd.Flags().String("algorandIndexerToken", "", "Algorand Indexer access token")
+	// algorandAlgodRPC = NodeCmd.Flags().String("algorandAlgodRPC", "", "Algorand Algod RPC URL")
+	// algorandAlgodToken = NodeCmd.Flags().String("algorandAlgodToken", "", "Algorand Algod access token")
+	// algorandAppID = NodeCmd.Flags().Uint64("algorandAppID", 0, "Algorand app id")
 
-	solanaWsRPC = NodeCmd.Flags().String("solanaWS", "", "Solana Websocket URL (required")
-	solanaRPC = NodeCmd.Flags().String("solanaRPC", "", "Solana RPC URL (required")
+	// solanaWsRPC = NodeCmd.Flags().String("solanaWS", "", "Solana Websocket URL (required")
+	// solanaRPC = NodeCmd.Flags().String("solanaRPC", "", "Solana RPC URL (required")
 
 	alphRPC = NodeCmd.Flags().String("alphRPC", "", "Alephium RPC URL (required)")
 	alphApiKey = NodeCmd.Flags().String("alphApiKey", "", "Alphium RPC api key")
 	alphContractIds = NodeCmd.Flags().StringSlice("alphContractIds", []string{}, "Alephium contract ids (required)")
 	alphGroupIndex = NodeCmd.Flags().Uint8("alphGroupIndex", 0, "The group index where contracts are deployed (required)")
 	alphMinConfirmations = NodeCmd.Flags().Uint8("alphMinConfirmations", 10, "The min confirmations for alephium tx")
+	alphFetchPeriod = NodeCmd.Flags().Uint8("alphFetchPeriod", 10, "The fetch events period for alwphium watcher")
 
 	logLevel = NodeCmd.Flags().String("logLevel", "info", "Logging level (debug, info, warn, error, dpanic, panic, fatal)")
 
+	devnetGuardianIndex = NodeCmd.Flags().Int("devnetGuardianIndex", 0, "Specify devnet guardian index")
+	integrationTest = NodeCmd.Flags().Bool("integrationTest", false, "Launch node for integration testing")
```

Comment:
Question: What is the purpose of this flag?

Reply:
These two flags are only used for integration tests:

* devnetGuardianIndex:
On devnet, the guardian key is automatically generated by the guardian index, and the node hostname is in the format:  `guardian-$index`, but on github CI, the hostname is random, so I added the `devnetGuardianIndex` flag. It first tries to get the index from the hostname, and if that fails, it gets the index from the `devnetGuardianIndex` flag.

* integrationTest
The guardian code will lock the memory to protect secret keys from being swapped out to disk, the code is [here](https://github.com/alephium/wormhole-fork/blob/6c1e42adb3b6d86f5afde55d7ee6501bdcc73af8/node/cmd/guardiand/node.go#L304), but github CI does not allow lock the memory, so I added the `integrationTest` flag to skip lock memory.

Reply:
Thanks! I was thinking about the `integrationTest` flag, now i got it. Thanks for the explanation :sun_with_face: